File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        vector<int> temp;
        ListNode* result = new ListNode();
        ListNode* tempNode = result;
        while(head){
            temp.push_back(head->val);
            tempNode->next = new ListNode(head->val);
            tempNode = tempNode->next;
            head = head->next;
        }
        tempNode = result;
        for(int i=0;i<left;i++){
            tempNode = tempNode->next;
        }
        for(int i=right;i>=left;i--){
            tempNode->val = temp[i-1];
            tempNode = tempNode->next;
        }
        return result->next;
    }
};

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        ListNode node = null;
        ListNode startNode = null, lastNode = null;
        int index = 1;
        ListNode cur = head, nodeCur = node;
        while(cur != null){
            if(index < left) startNode = cur;
            else if(index == left) {
                node = new ListNode(cur.val);
                lastNode = node;
            }
            else if(index > left && index <= right){
                ListNode temp = new ListNode(cur.val);
                temp.next = node;
                node = temp;
            }else if(index == right + 1){
                if(startNode != null)
                    startNode.next = node;
                else head = node;
                lastNode.next = cur;
            }
            index++;
            cur = cur.next;
        }
        if(left == 1 && index == right + 1) return node;
        else if(index == right + 1) {
            startNode.next = node;
            lastNode.next = cur;
        }
        return head;
    }
}